### PR TITLE
#20 more testing for Utilities.js

### PR DESCRIPTION
**Pull request by DDCreationStudios**
Added a semicolon to the utility.js file and wrote more tests for the test file.
I am not quite sure if you want to test the "convertParams" function like that, since I am still lacking to fully grasp the Router file. 😉 
I am open for suggestions 👐 🎉  

Also notice that we can now deliberately assign Reviewers on Git🎉 🎉  🎉 🎉

### CODE REVIEW

File: lib/Utilities.spec.js [hunk 1]
```diff
@@ -27,4 +27,38 @@ describe('Utilities', () => {
       Utilities.convertFloatParam('not a number', 'float_key');
     });
   });
+
+  it('should convert single parameter', () => {
```
Comment (lumio) [inline lib/Utilities.spec.js]: This test covers too many methods.
Separate this test into more it-statements.

File: lib/Utilities.spec.js [hunk 2]
```diff
@@ -27,4 +27,38 @@ describe('Utilities', () => {
...
+      [2.1, 'key']
+    );
+  });
+
+  it('should convert parameter into an object', () => {
+    assert.deepEqual(
+      Utilities.convertParams('child'),
+      {'0': 'c', '1': 'h', '2': 'i', '3': 'l', '4': 'd'}
+    );
+    assert.deepEqual(
```
Comment (lumio) [inline lib/Utilities.spec.js]: Same as above. You can use strings as arrays, which is useful sometimes, but not valid here.

File: lib/Utilities.spec.js [hunk 3]
```diff
@@ -27,4 +27,38 @@ describe('Utilities', () => {
...
+      Utilities.convertParams('2'),
+      {'0': '2'}
+    );
+    assert.deepEqual(
```
Comment (lumio) [inline lib/Utilities.spec.js]: `convertParams` should always return false when first parameter is not an array.

File: lib/Utilities.spec.js [hunk 4]
```diff
@@ -27,4 +27,38 @@ describe('Utilities', () => {
...
+      Utilities.convertParams(2),
+      {}
+    );
+    assert.isObject(
+      Utilities.convertParams(null||function(){}||undefined||true||NaN||2),
```
Comment (lumio) [inline lib/Utilities.spec.js]: This test doesn't do anything useful, cause you pass the first value that is equivalent to `true`. In this case the function.

The `||` means `OR` and just tests if one of those statements is equivalent to `true` and stops at the first one.

You can test this as follows:
```
var testOr = null || function() {} || undefined || true || NaN || 2;
console.log( typeof testOr );
```

Also watch your whitespaces: `null || function() {} || undefined || true || NaN || 2`

For readability you should write out your tests or use an array with testable objects.
Maybe we can talk about this at our next meeting, cause this is a great but also misleading behaviour of javascript.

File: lib/Utilities.js [hunk 5]
```diff
@@ -43,11 +43,9 @@ module.exports = {
       if ( !params.hasOwnProperty( key ) ) {
         continue;
       }
-
```
Comment (lumio) [inline lib/Utilities.js]: Please leave those newlines for readability.

File: lib/Utilities.js [hunk 6]
```diff
@@ -43,11 +43,9 @@ module.exports = {
...
       let result = this.convertSingleParam( params[ key ], key );
       newParams[ result[ 1 ] ] = result[ 0 ];
     }
-
```
Comment (lumio) [inline lib/Utilities.js]: Same as above 😄

File: lib/Utilities.js [hunk 7]
```diff
@@ -43,11 +43,9 @@ module.exports = {
...
     return newParams;
   },
-}
```
Comment (lumio) [inline lib/Utilities.js]: 👍

**Comment by lumio** [comment]
Maybe we should talk about those changes at our next meeting.

**Comment by DDCreationStudios** [comment]
Ok cool. Let's talk about that in the next meeting. I will leave it as it is now 😉

**Comment by lumio** [comment]
This isn't done yet 😉

**Comment by lumio** [comment]
See pull request 27 (https://github.com/cheesein/cheesein/pull/27)

**Pull request closed without merging**